support-frontend/app/controllers/DigitalSubscriptionController.scala

```diff
@@ -3,174 +3,84 @@ package controllers
-      "en" -> buildCanonicalDigitalSubscriptionLink("int", orderIsAGift)
-    )
-    val shareImageUrl = Some("https://i.guim.co.uk/img/media/74422ad120c709448f433c34f5190e2465ffa65e/0_0_1200_1200/1200.png?width=1200&auto=format&fit=crop&quality=85&s=1407add4d016d15cc074b0f9de8f1433") // scalastyle:ignore
-    if (settings.switches.enableDigitalSubGifting.isOn || !orderIsAGift) {
+
+    if (!settings.switches.enableDigitalSubGifting.isOn && orderIsAGift) {
+      Redirect(routes.DigitalSubscriptionController.digitalGeoRedirect(false)).withSettingsSurrogateKey
+    } else {
+      val canonicalLink = Some(buildCanonicalDigitalSubscriptionLink("uk", orderIsAGift))
+      val queryPromos = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil)
       Ok(views.html.main(
-        title = title,
-        mainElement = mainElement,
-        mainJsBundle = js,
-        mainStyleBundle = css,
+        title = s"Support the Guardian | The Guardian Digital ${if (orderIsAGift) "Gift " else ""}Subscription",
+        mainElement = EmptyDiv("digital-subscription-landing-page-" + countryCode),
+        mainJsBundle = Left(RefPath("digitalSubscriptionLandingPage.js")),
+        mainStyleBundle = Left(RefPath("digitalSubscriptionLandingPage.css")),
         fontLoaderBundle = fontLoaderBundle,
-        description = description,
+        description = stringsConfig.digitalPackLandingDescription,
         canonicalLink = canonicalLink,
-        hrefLangLinks = hrefLangLinks,
-        shareImageUrl = shareImageUrl,
+        hrefLangLinks = getPaperHrefLangLinks(orderIsAGift),
+        shareImageUrl = Some(
+          "https://i.guim.co.uk/img/media/74422ad120c709448f433c34f5190e2465ffa65e/0_0_1200_1200/1200.png" +
+            "?width=1200&auto=format&fit=crop&quality=85&s=1407add4d016d15cc074b0f9de8f1433"
+        ),
         shareUrl = canonicalLink
       ) {
-        Html(s"""<script type="text/javascript">
+        val maybePromotionCopy = landingCopyProvider.promotionCopy(queryPromos, DigitalPack, countryCode, DefaultPromotions.DigitalSubscription.landing)
+        Html(
+          s"""<script type="text/javascript">
           window.guardian.productPrices = ${outputJson(productPrices(queryPromos, orderIsAGift))}
-          window.guardian.promotionCopy = ${outputJson(landingCopyProvider.promotionCopy(queryPromos, countryCode))}
+          window.guardian.promotionCopy = ${outputJson(maybePromotionCopy)}
           window.guardian.orderIsAGift = $orderIsAGift
-        </script>""")
-      }).withSettingsSurrogateKey
-    } else {
-      Redirect(routes.DigitalSubscriptionController.digitalGeoRedirect(false)).withSettingsSurrogateKey
-    }
-  }
-
-  def digitalGeoRedirect(orderIsAGift: Boolean = false): Action[AnyContent] = geoRedirect(
-    if (orderIsAGift) "subscribe/digital/gift" else "subscribe/digital"
-  )
-
-  def displayForm(orderIsAGift: Boolean): Action[AnyContent] = {
```

Comment:
I've moved the displayForm into a separate controller becaase this Controller had a huge number of parmeters and there didn't seem to be much connection between the two sides other than they happen to be the same product flow.
I did the same for weekly and paper, for consistency.

Reply:
👏